src/store/middleware/model-poller.ts

```diff
@@ -440,6 +442,118 @@ export const modelPollerMiddleware: Middleware<
       // The action has already been passed to the next middleware
       // at the top of this handler.
       return;
+    } else if (
+      isSpecificAction<ReturnType<typeof jujuActions.destroyModels>>(
+        action,
+        jujuActions.destroyModels.type,
+      )
+    ) {
+      // Intercept destroyModel actions and fetch and store
+      // the modelParams via the controller connection.
+      const { wsControllerURL, modelParams } = action.payload;
+      // Immediately pass the action along so that it can be handled by the
+      // reducer to update the loading state.
+      next(action);
+      const conn = controllers.get(wsControllerURL);
+      if (!conn) {
+        return;
+      }
+      let remainingModels: string[] = [];
+
+      try {
+        const result = await conn.facades.modelManager?.destroyModels({
+          models: modelParams,
+        });
+        const errors: DestroyModelErrors = [];
+        result?.results.forEach((errorResult, index) => {
+          if (errorResult.error) {
+            errors.push([
+              modelParams[index]["model-tag"],
+              errorResult.error.message,
+            ]);
+          } else {
+            remainingModels.push(modelParams[index]["model-tag"]);
+          }
+        });
+
+        // Dispatch partial errors, if any, and continue with successful destructions
+        if (errors.length > 0) {
+          reduxStore.dispatch(jujuActions.destroyModelErrors({ errors }));
+        }
+
+        // Only proceed to check for completion if at least one model had no errors
+        if (errors.length < modelParams.length) {
```

Comment:
I think there needs to be a test that it does not poll if there are no successful requests.